Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -48,41 +64,82 @@ void APSStarActor::OnLocalCharacterReady(APlayerCharacter* Character, int32 Char
 	}
 }
 
+// Called when the current game state was changed
+void APSStarActor::OnGameStateChanged(ECurrentGameState GameState)
+{
+	if (AMyGameStateBase::GetCurrentGameState() == ECurrentGameState::Menu)
+	{
+		SetStartTimeMenuStars();
+		TryPlayMenuStarAnimation();
+	}
+	else
+	{
+		StartTimeMenuStarsInternal = 0.f;
+	}
+}
+
 // Is called when any cinematic started
 void APSStarActor::OnAnyCinematicStarted(const UObject* LevelSequence, const UObject* FromInstigator)
 {
-	const UWorld* World = GetWorld();
-	check(World);
-
-	StartTimeHideStarsInternal = World->GetTimeSeconds();
-
-	// Enable tick and process this frame
-	SetActorTickEnabled(true);
+	SetStartTimeHideStars();
 	TryPlayHideStarAnimation();
 }
 
-// Hiding stars with animation
+// Hiding stars with animation in main menu when cinematic is start to play
 void APSStarActor::TryPlayHideStarAnimation()
 {
-	if (!StartTimeHideStarsInternal)
+	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
+	const bool bIsFinished = !TryPlayStarAnimation(StartTimeHideStarsInternal, CurrentRow.HideStarsAnimation);
+	if (bIsFinished)
 	{
-		return;
+		StartTimeHideStarsInternal = 0.f;
+		UPoolManagerSubsystem::Get().ReturnToPool(this);
 	}
+}
 
+// Menu stars with animation in main menu idle 
+void APSStarActor::TryPlayMenuStarAnimation()
+{
 	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
-	if (!CurrentRow.HideStarsAnimation)
+	bool bIsFinished = !TryPlayStarAnimation(StartTimeMenuStarsInternal, CurrentRow.MenuStarsAnimation);
+	if (bIsFinished)
 	{
-		SetActorTickEnabled(false);
-		StartTimeHideStarsInternal = 0.f;
-		return;
+		StartTimeMenuStarsInternal = GetWorld()->GetTimeSeconds();
 	}
+}
 
-	const float SecondsSinceStart = GetWorld()->GetTimeSeconds() - StartTimeHideStarsInternal;
-
-	const bool bIsFinished = !UGameplayUtilsLibrary::ApplyTransformFromCurveTable(this, CurrentRow.HideStarsAnimation, SecondsSinceStart);
-	if (bIsFinished)
+// Static helper function that plays any given star animation from various places
+bool APSStarActor::TryPlayStarAnimation(UPARAM(ref) float& StartTimeRef, UCurveTable* AnimationCurveTable)
```

Comment:
Please remove the `UPARAM(ref)` from the `.cpp` file as it is unnecessary there. The `UPARAM(ref)` macro is only needed in the `.h` file within the `UFUNCTION` declaration to ensure proper reflection in Unreal Engine's Blueprint system.

#### Why the change is needed:
- `UPARAM(ref)` is used in the `.h` file to correctly inform Unreal Engine's reflection system. Including it in the `.cpp` file has no impact on functionality and is redundant.

```.cpp
bool APSStarActor::TryPlayStarAnimation(float& StartTimeRef, UCurveTable* AnimationCurveTable)
{
    // Function implementation here
}
```

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/8f3f1e6a7abaedb69b5a8783f6a04c78f962f3f4